<?xml version="1.0"?>
<launch>
    <arg name="config_file" default="$(find my_aruco)/config/simulation_example.yaml"/>

    <!-- ================================================================================== -->
    <param name="config_file" value="$(arg config_file)"/>
    <node pkg="my_aruco" type="aruco_detect_node" name="aruco_detect_node" output="screen" required="true"/>
    <node pkg="my_aruco" type="aruco_optim_node" name="aruco_optim_node" output="screen" />

    <!-- ground truth from gazebo    -->
    <node pkg="my_aruco" type="sim_groundtruth_publisher_node" name="sim_groundtruth_publisher_node">
        <param name="task" value="slamdog"/>
        <param name="frame_id" value="link_base"/>
        <param name="child_frame_id" value="link_trailer"/>
    </node>

    <!-- robot state to get linear velocity v and angular velocity w -->
    <node pkg="my_aruco" type="slamdog_state_publisher.py" name="sim_slamdog_state_publisher" required="true"/>
</launch>